sriov/main.go

```diff
@@ -79,6 +79,12 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return errors.New("IPAM plugin returned missing IPv4 config")
 	}
 
+	defer func() {
+		if err != nil {
+			ipam.ExecDel(n.IPAM.Type, args.StdinData)
+		}
+	}()
+
 	err = netns.Do(func(_ ns.NetNS) error {
```

Comment:
This will disregard error on L:89 which is IMO what we want to check most.
After getting an IP on L:73 we might fail on configuring the interface on L:89 and only then we should try to release the IP that was allocated. So, this defer block should be moved after L:90 here.

Reply:
if we failed on the move namespace we need to call ipam.ExecDel to return the ip address to the pool. 
you can see a good example with macvlan see [1]. 

[1] - https://github.com/containernetworking/plugins/blob/master/plugins/main/macvlan/macvlan.go#L173-L184

Reply:
I got that. My concern was if this defer block would get the errors coming from further down. I just double checked it again, see no issues.